### DISCUSSION TITLE
How to import "DisconnectDescription" for disconnect handler?

### DISCUSSION THREAD

**Discussion by apudiu** [socketio/socket.io#5221] [Q&A]
Hi, I'm seeing that the disconnect handler (for `socket.on('disconnect', handlerFn)`  is of type `(reason: Socket.DisconnectReason, description?: DisconnectDescription) => void`. How to import the `DisconnectDescription` type? 
My IDE did following
```typescript
import {DisconnectDescription} from "socket.io-client/build/esm-debug/socket";
```
Which is raising a TS error
```log
TS2307: Cannot find module socket.io-client/build/esm-debug/socket or its corresponding type declarations.
```
Now what to do? I think as the `DisconnectReason` is inside the `Socket`, why not the `DisconnectDescription` is in there?

i'm importing `Socket` from
```typescript
import {io, Socket} from "socket.io-client";
```

**Comment by daffyzk** [comment]
same for me :(

if I try to do this:

` details?: Socket.DisconnectDescription | undefined`

the error I get is:

`Namespace '"/.../node_modules/socket.io-client/build/esm/socket".Socket' has no exported member 'DisconnectDescription'.ts(2694)`

even though it's exported inside the socket.d.ts file (line 51):

`export type DisconnectDescription = Error | {
    description: string;
    context?: unknown;
};`

**Comment by ajesquirell** [comment]
Also wondering...

**Comment by marcel-lindenbergh-spotler** [comment]
I am running into this issue too now after upgrading to Angular v21. Anyone?

**Comment by darrachequesne** [comment]
Hi! `DisconnectDescription` is now properly exported: https://github.com/socketio/socket.io/commit/98741e15e958a86bf8d159b9c551138d55551a27

Included in `socket.io-client@4.8.2`.
